### PR TITLE
002 Script for requirement [PerformAudioPassThru] SDL must transfer…

### PR DESCRIPTION
**Pull request by maria-mih**
… request to HMI in case of valid audioPassThruIcon param

002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua 
All common functions used are available at #703
Testing coverage available at #643
@istoimenova Please review

### CODE REVIEW

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+-- [HMI_API] [MOBILE_API] The "audioPassThruIcon" param at "ImageFieldName" struct
+--
+-- Description:
+-- In case mobile app sends PerformAudioPassThru_request to SDL with:
+-- valid <audioPassThruIcon> parameter
+-- and the requested <audioPassThruIcon> exists at app`s sandbox (see AppStorageFolder section)
+-- as well as another related to request valid params
+-- SDL must transfer UI.PerformAudioPassThru (<audioPassThruIcon>, other params)_request + Speak_request (depends on parameters provided by the app) to HMI
+--1. Used preconditions
+--1. All params used in PerformAudioPassThru_request are present and in upper bound. Please note that for parmeter <<value>> from image parameter the actual limitation in size is 255 symbols (Actual Ubuntu limitation for file name length)
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih, 
suggestion: to be more readable, such long comments may be split in few rows.

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+--1. Used preconditions
+--1. All params used in PerformAudioPassThru_request are present and in upper bound. Please note that for parmeter <<value>> from image parameter the actual limitation in size is 255 symbols (Actual Ubuntu limitation for file name length) 
+--2. AudioPassThruIcon exists at apps sub-directory of AppStorageFolder (value from ini file)
+--2. Performed steps
+-- Send PerformAudioPassThru (audioPassThruIcon, other params) from mobile to SDL and check:
+--2.1 SDL sends UI.PerformAudioPassThru (audioPassThruIcon, other params) to HMI
+--2.2 SDL sends TTS.Speak to HMI
+--2.3 HMI sends UI.PerformAudioPassThru (SUCCESS) to SDL
+--2.4 HMI sends TTS.Speak (SUCCESS) to SDL
+-- Expected result:
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih , expected result should be:
SDL sends UI.PerformAudioPassThru (audioPassThruIcon, other params) to HMI
SDL sends TTS.Speak to HMI

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+-- as well as another related to request valid params
+-- SDL must transfer UI.PerformAudioPassThru (<audioPassThruIcon>, other params)_request + Speak_request (depends on parameters provided by the app) to HMI
+--1. Used preconditions
+--1. All params used in PerformAudioPassThru_request are present and in upper bound. Please note that for parmeter <<value>> from image parameter the actual limitation in size is 255 symbols (Actual Ubuntu limitation for file name length) 
+--2. AudioPassThruIcon exists at apps sub-directory of AppStorageFolder (value from ini file)
+--2. Performed steps
+-- Send PerformAudioPassThru (audioPassThruIcon, other params) from mobile to SDL and check:
+--2.1 SDL sends UI.PerformAudioPassThru (audioPassThruIcon, other params) to HMI
+--2.2 SDL sends TTS.Speak to HMI
+--2.3 HMI sends UI.PerformAudioPassThru (SUCCESS) to SDL
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih, points 2.3-2.4 are redundant in description in scope of this requirement.

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+commonSteps:PutFile("Precondition_PutFile_With_Icon", string.rep ("a", 251).. ".png")
+
+function Test:Precondition_Check_audioPassThruIcon_Existence()
+  testCasesForPerformAudioPassThru:Check_audioPassThruIcon_Existence(self)
+end
+
+function Test:Precondition_ActivateApp()
+  testCasesForPerformAudioPassThru:ActivateAppDiffPolicyFlag
+  (self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih ,please use parameters of function on the same way

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+function Test:TestStep_PerformAudioPassThru_AllParameters_Upper_SUCCESS()
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih , please be sure that RPC PerformAudioPassThru is allowed by policy.

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  testCasesForPerformAudioPassThru:PerformAudioPassThru_AllParameters_Upper_SUCCESS (self)
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih, as the idea of atomic script is to show what exactly is happening, will be better to declare the TestStep in script, not to call common function.
Calling of common functions is useful as preconditions/postconditions.

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local testCasesForPerformAudioPassThru = require('user_modules/shared_testcases/testCasesForPerformAudioPassThru')
+
+--[[ General Precondition before ATF start ]]
+commonSteps:DeleteLogsFiles()
+commonSteps:DeletePolicyTable ()
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih, please remove space before brackets. Do the same for all calling of functions.

**Comment by istoimenova** [comment]
@maria-mih, please reference script in the TestSet.

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 1 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @istoimenova  Done in commit 6e96740c09fa54b1c5c7d16430e95877fd8fa1fb

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 3 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @istoimenova  Updated in commit 6e96740c09fa54b1c5c7d16430e95877fd8fa1fb

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 2 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @istoimenova  Updated in commit 6e96740c09fa54b1c5c7d16430e95877fd8fa1fb

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 7 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @istoimenova  Done in commit 6e96740c09fa54b1c5c7d16430e95877fd8fa1fb

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 4 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @istoimenova  Done in commit 6e96740c09fa54b1c5c7d16430e95877fd8fa1fb

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 5 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @istoimenova  Check added in commit 6e96740c09fa54b1c5c7d16430e95877fd8fa1fb

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 6 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @istoimenova  Done in commit 6e96740c09fa54b1c5c7d16430e95877fd8fa1fb

**Comment by maria-mih** [comment]
Script has been updated after first review iteration. Please check on commit  6e96740c09fa54b1c5c7d16430e95877fd8fa1fb

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 1 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih, ACCEPTED.

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 3 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih, ACCEPTED.

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 2 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih, ACCEPTED.

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 7 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih, ACCEPTED.

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 4 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih, ACCEPTED.

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 5 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih, ACCEPTED.

File: test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua (hunk 6 above)
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/002_ATF_P_PerformAudioPassThru_SUCCESS_All_Params_Upper.lua]: @maria-mih, ACCEPTED.

**Pull request merged**